controllers/base_controller.go

```diff
@@ -108,6 +108,12 @@ func (r *BaseReconciler) getSMClient(ctx context.Context, object api.SAPBTPResou
 		}
 	}
 
+	if len(cfg.ClientID) == 0 ||
+		len(cfg.ClientSecret) == 0 ||
```

Comment:
@kerenlahav this condition would always result in showing this error message `invalid Service-Manager credentials, contact your cluster administrator` when you don't create clientID and clientSecret servicekey but TLS based service key. In case of TLS based service key the `cfg.ClientSecret` is always empty.